File: PostAsync.py
from flask import Flask, request, jsonify
import requests
import base64
import os

app = Flask(__name__)

# Your GitHub config
GITHUB_TOKEN = os.getenv("GITHUB_TOKEN") or "YOUR_GITHUB_PAT"
OWNER = "iNeedFreedom"
REPO = "FixCarLimited"
FILE_PATH = "Data.js"

# Get file info from GitHub (sha required for updates)
def get_file_info():
    url = f"https://api.github.com/repos/{OWNER}/{REPO}/{FILE_PATH}"
    headers = {
        "Authorization": f"token {GITHUB_TOKEN}",
        "User-Agent": "Python-App"
    }
    resp = requests.get(url, headers=headers)
    if resp.status_code == 200:
        return resp.json()
    else:
        print("Failed to fetch file:", resp.text)
        return None

@app.route("/roblox", methods=["POST"])
def roblox_post():
    data = request.get_json()
    if not data:
        return jsonify({"error": "No JSON received"}), 400

    # Encode new JSON content
    new_content = base64.b64encode(
        bytes(str(data).replace("'", '"'), "utf-8")
    ).decode("utf-8")

    # Get current sha
    file_info = get_file_info()
    if not file_info:
        return jsonify({"error": "Could not fetch file"}), 500

    sha = file_info["sha"]

    # Prepare request body
    body = {
        "message": "Update Data.js from Roblox",
        "content": new_content,
        "sha": sha
    }

    url = f"https://api.github.com/repos/{OWNER}/{REPO}/contents/{FILE_PATH}"
    headers = {
        "Authorization": f"token {GITHUB_TOKEN}",
        "User-Agent": "Python-App",
        "Content-Type": "application/json"
    }

    resp = requests.put(url, json=body, headers=headers)

    if resp.status_code in [200, 201]:
        return jsonify({"status": "success", "github": resp.json()})
    else:
        return jsonify({"error": resp.text}), resp.status_code

if __name__ == "__main__":
    app.run(host="0.0.0.0", port=3000)

File: PostAsync.lua
local https = require("ssl.https")
local ltn12 = require("ltn12")
local cjson = require("cjson")
local mime = require("mime") -- for base64 encoding

local token = "BREADTOKEN33" -- your GitHub personal access token
local owner = "iNeedFreedom"
local repo = "FixCarLimited"
local path = "Data.js"

-- Step 1: Fetch current file SHA
local response_body = {}
local _, code = https.request{
    url = "https://api.github.com/repos/"..owner.."/"..repo.."/contents/"..path,
    method = "GET",
    headers = {
        ["Authorization"] = "token " .. token,
        ["User-Agent"] = "Lua-App"
    },
    sink = ltn12.sink.table(response_body)
}

if code ~= 200 then
    print("Failed to fetch file, code:", code)
    print(table.concat(response_body))
    return
end

local fileData = cjson.decode(table.concat(response_body))
local sha = fileData.sha

-- Step 2: Prepare new content
local newData = { CurrentCar = 2 }
local newJson = cjson.encode(newData)
local base64Content = mime.b64(newJson)

local updateBody = cjson.encode({
    message = "Update Data.js from Lua",
    content = base64Content,
    sha = sha
})

-- Step 3: Send PUT request to update file
local updateResponse = {}
local _, updateCode = https.request{
    url = "https://api.github.com/repos/"..owner.."/"..repo.."/contents/"..path,
    method = "PUT",
    headers = {
        ["Authorization"] = "token " .. token,
        ["User-Agent"] = "Lua-App",
        ["Content-Type"] = "application/json",
        ["Content-Length"] = tostring(#updateBody)
    },
    source = ltn12.source.string(updateBody),
    sink = ltn12.sink.table(updateResponse)
}

local lapis = require("lapis")
local app = lapis.Application()

app:post("/roblox", function(self)
  local data = self.params  -- Roblox JSON body will appear here
  print("Got data from Roblox:", require("cjson").encode(data))
  return { json = { status = "ok", received = data } }
end)

return app
